app/routes/products_.$productId/route.tsx

```diff
@@ -41,8 +42,16 @@ export default function ProductDetailsPage() {
         if (!product?._id) {
             return;
         }
-        const quantity = parseInt(quantityInput.current?.value || '1', 10);
 
+        setAddToCartAttempted(true);
+        if (
```

Comment:
let's just set proper initial value for the `selectedOptions` state (I mean `{size: null, color: null, ...}`) and here just check for `Object.values(selectedOptions).includes(null)`